### ISSUE TITLE
Checking for absent views

### ISSUE THREAD

**Issue by cameroncooke** [nalexn/ViewInspector#129]
I've created a function to aid determining if a view is absent to reduce boilerplate in test cases:

```swift
func isViewAbsent<ViewType>(_ closure: @autoclosure () throws -> InspectableView<ViewType>) -> Bool {
    do {
        _ = try closure()
    } catch InspectionError.viewNotFound {
        return true
    } catch {}

    return false
}
```

Used like:

```swift
func testButtonIsShown() throws {

        let sut = MyView()
        XCTAssertTrue(isViewAbsent(try sut.inspect().button()))
}
```

This is useful when you've got views like:

```swift
struct MyView: View {
    
    @State var showButton = false

    var body: some View {
         if showButton {
             Button(....)
         }
        ....
     }
}
```

Is there a simpler approach, I was hoping for something like:

```swift
try sut.inspect().find(BackButton.self).button().isAbsent()
```

But couldn't find anything. Would be a useful method to have in the library?

**Comment by nalexn** [comment]
Hm, I'm usually using `XCTAssertThrowsError(try ...)`. Not super precise, as the expression may throw earlier and give a false positive (passing test when the entire `BackButton` is missing, for example).
I don't think Swift allows exception handling in the way you proposed: when `.button()` throws the subsequent function call is aborted by design. So wrapping the expression is the only way.

**Comment by gedaiu** [comment]
I wrote a small extension which could be included in the library

```swift

extension View where Self : ViewInspector.Inspectable {
    func canFindView<V>(_ inspectable: V.Type) -> Bool where V : Inspectable {
        var found: Bool = true
        do {
            let _ = try self.inspect().find(inspectable)
        }
        catch {
            found = false
        }

        return found
    }
}

```

and use it like this:
```
    let found = subject.canFindView(FieldLongText.self)
    expect(found).to(equal(true))
```

**Comment by nalexn** [comment]
Checking for absent views is now available in v0.9.1, [an example](https://github.com/nalexn/ViewInspector/issues/136#issuecomment-974889914).
